src/fs/copy.js

```diff
@@ -1,5 +1,46 @@
+import fs from 'node:fs/promises';
+import { fileURLToPath } from 'node:url';
+import path, { dirname } from 'node:path';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
 const copy = async () => {
-    // Write your code here 
+  try {
+    const srcDirPath = path.join(__dirname, 'files');
+    const dstDirPath = path.join(__dirname, 'files_copy');
+
+    try {
+      await fs.access(srcDirPath);
+    } catch (error) {
+      throw new Error('FS operation failed');
+    }
+
+    try {
+      await fs.access(dstDirPath);
+      throw new Error('FS operation failed');
```

Comment:
No need for error throw.

Reply:
The error is thrown by inner try...catch when it checks the existence of files